### PR TITLE
Fixes #428: Implement `FakeFS::File#string` for `CSV.read` in Ruby v2.6.x 

### LINKED ISSUE #428: `FakeFS` and `CSV` issues in Ruby v2.6.x
Hey y'all,

Thanks for the great library!

---

Consider the following code:

```ruby
absolute_path_to_local_csv = "/path/to/file.csv"

FakeFS do
  FakeFS::FileSystem.clone(absolute_path_to_local_csv)
  CSV.read(absolute_path_to_local_csv)
end
```

In Ruby v2.3.x - v2.5.x, a nested array is returned such that each value is an array of columns values representing each row in the CSV (expected behavior).

However, after upgrading to Ruby v2.6.0, the same code produces this error:

```bash
undefined method `string' for #<FakeFS::File:0x00007fdb072b73e0>
```

I've tested this with Ruby v2.6.0 and v2.6.1, using multiple versions of `FakeFS`, from v0.11.0 to v0.19.1.

As far as I can tell, both `CSV.open` and `CSV.foreach` are effected by this, but I would assume `CSV.open` as well as other class methods may result in the same error.

---

I noticed that this may be related to a much older issue #192 (closed). Please let me know if I can provide any additional information to help debug this!

### PR DESCRIPTION
**Pull request by cbortz**
I've also added ruby v2.6.1 as a version to be built in TravisCI.

---

fixes #428

### CODE REVIEW

File: test/fixtures/file.csv [hunk 1]
```diff
@@ -0,0 +1,27 @@
+headercolumn_0,headercolumn_1,headercolumn_2,headercolumn_3,headercolumn_4,headercolumn_5,headercolumn_6,headercolumn_7,headercolumn_8,headercolumn_9
```
Comment (grosser) [inline test/fixtures/file.csv]: can we make this simpler ?

File: test/fakefs_file_test.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
+require_relative 'test_helper'
```
Comment (grosser) [inline test/fakefs_file_test.rb]: is there no other test that already covers file.rb ?

Comment (cbortz) [reply test/fakefs_file_test.rb]: I saw an explicit test file for `FaksFS::FakeFile` but not `FakeFS::File`. Any idea where it might be?

Comment (cbortz) [reply test/fakefs_file_test.rb]: Ah, yea it seems that some specs are present in the `FakeFSTest`. Would you prefer I move this test into that file (`/test/fakefs_test.rb`)?

File: test/fixtures/file.csv (hunk 1 above)
Comment (cbortz) [reply test/fixtures/file.csv]: How do you mean? I'm happy to make the file more reasonable. My intent is to ensure we have enough data in the file to raise the unwanted exception.

Any suggestions?

File: test/fakefs_file_test.rb (hunk 2 above)
Comment (grosser) [reply test/fakefs_file_test.rb]: yeah, prefer to have 1 file then piecemeal ... ideally extract all the file tests to a single test-file

Comment (cbortz) [reply test/fakefs_file_test.rb]: > ideally extract all the file tests to a single test-file

Will do!

**Comment by cbortz** [comment]
@grosser I've fixed up the second commit to change two things:

1. Attempt to simplify the actual CSV generation
    + Remove the fixture CSV
    + Generate a logical CSV at runtime that should be 1025 bytes (1 above the chunk threshold)
2. Move the spec into the FakeFS test (`test/fakefs_test.rb`)
    + I wanted to move the `File` specs out
    + However, they are sprinkled throughout the test file and share helpers
        * I think this type of re-structuring may be out-of-scope for this PR, however
        * I'd be more than happy to issue another PR to restructure the test files

---

Please have a look 🙏

File: test/fakefs_test.rb [hunk 3]
```diff
@@ -2042,6 +2045,23 @@ def test_is_not_a_stringio
...
   end
 
+  def test_can_be_read_via_csv_library
+    csv_rows = [
+      Array.new(171) { '1' }.join(','), # 341 bytes
+      Array.new(171) { '2' }.join(','), # 341 bytes
+      Array.new(171) { '3' }.join(','), # 341 bytes
+    ]
+
+    csv_string = csv_rows.join("\n") # 1025 total bytes = (341 characters * 3) + (2 newline characters)
```
Comment (grosser) [inline test/fakefs_test.rb]: add a comment on why this exact number of bytes is important

**Comment by grosser** [comment]
perfect!
restructure would be nice, but yes it should be it's own clean PR :)

File: test/fakefs_test.rb [hunk 4]
```diff
@@ -2024,8 +2025,10 @@ def self.omitted_file_methods
     end
   end
 
-  def test_file_should_not_respond_to_string_io_unique_methods
-    uniq_string_io_methods = StringIO.instance_methods - RealFile.instance_methods
+  def test_file_should_not_respond_to_string_io_unique_methods_except_string
+    uniq_string_io_methods = (StringIO.instance_methods - RealFile.instance_methods)
+    uniq_string_io_methods.delete(:string)
```
Comment (grosser) [inline test/fakefs_test.rb]: add comment why we need to delete this

**Comment by grosser** [comment]
I'm still a bit confused by why we need the exact byte count

**Comment by cbortz** [comment]
> I'm still a bit confused by why we need the exact byte count

@grosser I added a comment explaining, which I will share here:

> Changes to how CSV's are read were introduced in Ruby 2.6.x. When parsing the CSV, `CSV::Parser` will break any non `File` object into chunks of 1024 bytes. If there's more than one chunk (i.e. file > 1024 bytes), each chunk will have `#string` invoked.
>
> In this spec, we generate a `FakeFS::File` with 1025 bytes of data, which pushes us just over the threshold of 1024 bytes. This provides an adequate test setup.
>
> We could make this number much higher than 1025. The only thing that matters is that the file is larger than 1024 bytes.

ℹ️ For reference:

- [`CSV::Parser#resolve_row_separatorL404`](https://github.com/ruby/csv/blob/v3.0.4/lib/csv/parser.rb#L404) - chunking logic
- [`CVS::Parser::InputsScanner#read_chunkL138`](https://github.com/ruby/csv/blob/v3.0.4/lib/csv/parser.rb#L138) - invocation of `#string` (**source of bug**)

File: test/fakefs_test.rb (hunk 4 above)
Comment (cbortz) [reply test/fakefs_test.rb]: Done!

File: test/fakefs_test.rb (hunk 3 above)
Comment (cbortz) [reply test/fakefs_test.rb]: Done!

**Comment by grosser** [comment]
v0.19.2

**Pull request merged**